src/main/kotlin/no/nav/syfo/infrastructure/journalforing/JournalforingService.kt

```diff
@@ -15,13 +21,61 @@ class JournalforingService(private val dokarkivClient: DokarkivClient, private v
             navn = navn,
             pdf = pdf
         )
+
+        journalfor(journalpostRequest)
+    }
+
+    override suspend fun journalfor(personident: Personident, behandlermelding: Behandlermelding, pdf: ByteArray): Result<JournalpostId> = runCatching {
+        val behandler = dialogmeldingBehandlerClient.getBehandler(behandlermelding.behandlerRef) ?: throw RuntimeException("Failed to get behander from isdialogmelding")
+        val behandlerFnr = behandler.fnr ?: throw RuntimeException("Behandler missing fnr")
+        val behandlerNavn = behandler.fullName()
+
+        val journalpostRequest = createJournalpostRequest(personident, behandlermelding, pdf, behandlerFnr, behandlerNavn)
+
+        journalfor(journalpostRequest)
+    }
+
+    private suspend fun journalfor(journalpostRequest: JournalpostRequest): JournalpostId {
         val journalpostId = dokarkivClient.journalfor(journalpostRequest).journalpostId
 
-        JournalpostId(journalpostId.toString())
+        return JournalpostId(journalpostId.toString())
     }
 
-    override suspend fun journalfor(personident: Personident, behandlermelding: Behandlermelding, pdf: ByteArray): Result<JournalpostId> {
-        TODO("Not yet implemented")
+    private fun createJournalpostRequest(
+        personident: Personident,
+        behandlermelding: Behandlermelding,
+        pdf: ByteArray,
+        behandlerFnr: String,
+        behandlerNavn: String
+    ): JournalpostRequest {
+        val avsenderMottaker = AvsenderMottaker.create(
+            id = behandlerFnr,
+            idType = BrukerIdType.PERSON_IDENT,
+            navn = behandlerNavn,
+        )
+        val bruker = Bruker.create(
+            id = personident.value,
+            idType = BrukerIdType.PERSON_IDENT,
+        )
+
+        val tittel = "Melding til behandler om friskmelding til arbeidsformidling"
+        val dokumenter = listOf(
+            createPdfDokument(
+                pdf = pdf,
+                brevkode = BrevkodeType.BEHANDLERMELDING_FRISKMELDING_TIL_ARBEIDSFORMIDLING,
+                tittel = tittel
+            ),
+        )
+
+        return JournalpostRequest(
+            avsenderMottaker = avsenderMottaker,
+            tittel = tittel,
+            bruker = bruker,
+            dokumenter = dokumenter,
+            kanal = JournalpostKanal.HELSENETTET.name,
+            overstyrInnsynsregler = OverstyrInnsynsregler.VISES_MASKINELT_GODKJENT.name,
```

Comment:
Betyr at dokumenter blir synlig for bruker på innloggede sider (tilsvarende slik vi har det for de fleste dialogmeldingene som sendes fra isbehandlerdialog)